examples/llm/src/models/layers/gpt_blocks.py

```diff
@@ -56,3 +56,39 @@ def forward(
         n = self.mlp(m)
         x = x + self.resid_mlp_dropout(n)
         return x
+
+
+class GPTBlockPostLN(nn.Module):
+
+    def __init__(self,
+                 cfg: DictConfig,
+                 causal_attn_cls: nn.Module,
+                 device: Optional[str] = None):
+        super().__init__()
+        if cfg.get('alibi', False):
+            assert cfg.attn_impl == 'triton' or cfg.attn_impl == 'torch', 'Only triton kernel or torch supports alibi'
+        self.ln_1 = nn.LayerNorm(cfg.d_model, device=device)
+        self.causal_attn = causal_attn_cls(cfg, device)
+        self.ln_2 = nn.LayerNorm(cfg.d_model, device=device)
+        self.mlp = GPTMLP(cfg, device=device)
```

Comment:
```suggestion
        self.causal_attn = causal_attn_cls(cfg, device)
        self.ln_1 = nn.LayerNorm(cfg.d_model, device=device)
        self.mlp = GPTMLP(cfg, device=device)
        self.ln_2 = nn.LayerNorm(cfg.d_model, device=device)

```

nit: I like to init things in the order they are applied.